modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -3640,7 +3668,7 @@ private void failIfSchemaChangedSinceTxStart(UUID txId, HybridTimestamp operatio
     }
 
     private CompletableFuture<Integer> reliableCatalogVersionFor(HybridTimestamp ts) {
-        return schemaSyncService.waitForMetadataCompleteness(ts)
+        return waitForMetadataCompleteness(ts)
                 .thenApply(unused -> catalogService.activeCatalogVersion(ts.longValue()));
     }
 
```

Comment:
Switch from raft to replica is missing. I'm also mentioned it in the comment above within the context of decorated raft client.

Reply:
I don't get this one. `waitForMetadataCompleteness()` method guarantees to return a future which dependents will be executed in the partition-operations thread. How does it relate to a Raft client?

Reply:
I mean that raft response will be returned in messaging thread and we also won't to switch response processing to partition one.